nemo/collections/nlp/data/token_classification/token_classification_descriptor.py

```diff
@@ -34,16 +34,25 @@ def __init__(self, data_dir: str, modes: List[str] = ['train', 'test', 'dev'], p
             modes: list of the modes to read, it can be from ["train", "test", "dev"] by default.
             It is going to look for the data files at {data_dir}/{mode}.txt
         """
-        self.data_dir = data_dir
+        if not os.path.exists(data_dir):
+            raise FileNotFoundError(
+                "Dataset not found. For NER, CoNLL-2003 dataset can be obtained at "
+                "https://github.com/kyzhouhzau/BERT-NER/tree/master/data."
+            )
 
+        self.data_dir = data_dir
+        # when the model is restored from .nemo file, data might be missing, and data_desc values should be skipped
+        self.data_found = False
         unique_labels = set()
+
         for mode in modes:
             all_labels = []
             label_file = os.path.join(data_dir, 'labels_' + mode + '.txt')
```

Comment:
Why not use register artifact here to serialize the label file too and use return value to test for if it exists or not ? Then data fiu d flag becomes unnecessary

Reply:
the label file could be huge

Reply:
Ah makes sense